packages/flutter/lib/src/painting/basic_types.dart

```diff
@@ -197,6 +197,21 @@ enum AxisDirection {
   left,
 }
 
+/// An orientation along either the horizontal or vertical [Axis].
+enum AxisOrientation {
```

Comment:
It's not affected by TextDirection per se, as the whole point of adding this property was to be able to define the orientation of the scrollbar irrespective of the textDirection(for example a left scrollbar orientation for TextDirection.ltr). But, this property does limit the use of left, right orientation only for vertical scroll and top, bottom orientation only for horizontal scroll.

Reply:
Thanks for explaining! I am wondering if this should just be specific to the Scrollbar, and added as an enum in `scrollbar.dart`. 
Currently, this is only used there, so If we need it elsewhere in the framework, we can move it out of scrollbar later.  We might like to use a different name too, like ScrollbarOrientation or something along those lines.
This is because 
1. I don't know that it is readily apparent how this is different from `AxisDirection` above. The values are the same, and it may be confusing and lead to bugs from people implementing these differently. And
2. the assertions you mentioned. These are currently reliant on the user to enforce, so someone could implement an AxisOrientation without regard for whether or not we are scrolling vertically or horizontally. It might actually be nice to create a debug method to check that, or maybe make this a getter based on the current Axis to protect it from incorrect use.

WDYT? :)

Reply:
Hmm, that seems reasonable. About that getter, could you clarify if my approach is correct? The getter would check if the current Axis is vertical or horizontal, then
1.) It would assert if orientation is left or right, and Axis is horizontal.
2.) It would assert if orientation is top or bottom, and Axis is vertical.
If the above two assertions pass, it would return the `ScrollbarOrientation` which was passed in the scrollbar.

Reply:
I think that makes sense the way I am understanding it. :)

> Also if the getter already checks for it, should the assertions that I added in switch case be removed then?

You could keep them there and instead wrap this all up in a debug method if you want to make it reusable. :)

Reply:
Updated with a debug method :).